src/app/commands/poap/POAP.ts

```diff
@@ -137,20 +194,25 @@ module.exports = class poap extends SlashCommand {
 				command = SchedulePOAP(ctx, guildMember, ctx.options.schedule['mint-copies']);
 				break;
 			case 'start':
-				command = StartPOAP(ctx, guildMember, ctx.options.start.event, ctx.options.start['duration-minutes']);
+				command = StartPOAP(ctx, guildMember, ctx.options.start['platform'], ctx.options.start.event, ctx.options.start['duration-minutes']);
 				break;
 			case 'end':
-				command = EndPOAP(guildMember, ctx);
+				command = EndPOAP(guildMember, ctx.options.end['code'], ctx);
 				break;
 			case 'distribute':
-				command = DistributePOAP(ctx, guildMember, ctx.options.distribute['event']);
+				command = DistributePOAP(ctx, guildMember, ctx.options.distribute['event'], ctx.options.distribute['code']);
 				break;
+			case 'claim':
+				command = ClaimPOAP(ctx, guildMember, ctx.options.platform, ctx.options.claim['code']);
+				return;
 			default:
 				return ctx.send(`${ctx.user.mention} Please try again.`);
 			}
-			return this.handleCommandError(ctx, command);
+			this.handleCommandError(ctx, command);
+			return;
 		} catch (e) {
 			LogUtils.logError('failed to process POAP command', e);
+			return ctx.send('Sorry something is not working and our devs are looking into it.');
```

Comment:
'Welp, something is definitely broken. I would blame you, but I know better. I\'ll let my devs know something is wrong.'